pkg/kn/commands/revision/revision_list.go

```diff
@@ -80,6 +84,28 @@ func NewRevisionListCommand(p *commands.KnParams) *cobra.Command {
 					return nil
 				}
 			}
+
+			// sort revisionList by configuration generation key
+			sort.SliceStable(revisionList.Items, func(i, j int) bool {
+				a := revisionList.Items[i]
+				b := revisionList.Items[j]
+
+				// Convert configuration generation key from string to int for avoiding string comparison.
+				agen, err := strconv.Atoi(a.Labels[serving.ConfigurationGenerationLabelKey])
+				if err != nil {
+					fmt.Fprintf(cmd.OutOrStdout(), "Invalid type of configuration generation key: %s\n", err)
```

Comment:
not sure anymore whether we should print out an error here (its probably very confusing to the user and can't really correlated that to the `kn revision list` command he called).

I think its better that in case of an error you just return `a.Name < b.Name` silently.